autoscale-core/src/main/java/com/hpe/caf/autoscale/core/ScalerThread.java

```diff
@@ -266,4 +290,36 @@ private int establishMemLimitReached(final double currentMemoryLoad)
         }
         return 0;
     }
+
+    private boolean shouldBackoff()
+    {
+        boolean shouldBackoff = false;
+        if (backoff) {
+            backoffCount++;
+            switch (lastOperation) {
+                case SCALE_DOWN: {
+                    shouldBackoff = scaleUpBackoffAmount == -1 ? defaultBackoff() : backoffCount < scaleDownBackoffAmount;
+                    break;
+                }
+                case SCALE_UP: {
+                    shouldBackoff = scaleUpBackoffAmount == -1 ? defaultBackoff() : backoffCount < scaleUpBackoffAmount;
+                    break;
+                }
+                default: {
+                    shouldBackoff = defaultBackoff();
+                    break;
+                }
+            }
+        }
+        if (shouldBackoff) {
+                    backoff = false;
```

Comment:
I don't get it.  Why would you be setting `backoff` to `false` when `shouldBackoff` is `true`?  Won't this stop the backing off?